Mage.Sets/src/mage/cards/e/EcologicalAppreciation.java

```diff
@@ -0,0 +1,148 @@
+package mage.cards.e;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.ExileSpellEffect;
+import mage.cards.*;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreatureCard;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetOpponent;
+
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class EcologicalAppreciation extends CardImpl {
+
+    public EcologicalAppreciation(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{X}{2}{G}");
+
+        // Search your library and graveyard for up to four creature cards with different names that each have mana value X or less and reveal them. An opponent chooses two of those cards. Shuffle the chosen cards into your library and put the rest onto the battlefield. Exile Ecological Appreciation.
+        this.getSpellAbility().addEffect(new EcologicalAppreciationEffect());
+        this.getSpellAbility().addEffect(ExileSpellEffect.getInstance());
+    }
+
+    private EcologicalAppreciation(final EcologicalAppreciation card) {
+        super(card);
+    }
+
+    @Override
+    public EcologicalAppreciation copy() {
+        return new EcologicalAppreciation(this);
+    }
+}
+
+class EcologicalAppreciationEffect extends OneShotEffect {
+
+    EcologicalAppreciationEffect() {
+        super(Outcome.Benefit);
+        staticText = "search your library and graveyard for up to four creature cards with different names that each have mana value X or less and reveal them. An opponent chooses two of those cards. Shuffle the chosen cards into your library and put the rest onto the battlefield";
+    }
+
+    private EcologicalAppreciationEffect(final EcologicalAppreciationEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public EcologicalAppreciationEffect copy() {
+        return new EcologicalAppreciationEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        int xValue = source.getManaCostsToPay().getX();
+        TargetCard targetCardsInLibrary = new EcologicalAppreciationTarget(Zone.LIBRARY, 4, xValue);
+        boolean searched = player.choose(Outcome.PutCreatureInPlay, new CardsImpl(player.getLibrary().getCards(game)), targetCardsInLibrary, game);
```

Comment:
can use `player.searchLibrary(...)` instead if desired -- went with this approach to avoid having to make two kinds of targets for `EcologicalAppreciationTarget`: one for `TargetCardInLibrary` and one for `TargetCardInGraveyard`

Reply:
Simple xmage hint:
* If you can use existed classes and methods then use it;
* If you need to create a new one then make research and ensure that you really need it;
* Custom classes (card's specific classes) are tech debt and must be skipped as much as possible.

Reply:
understood 👍